src/__tests__/sharedFunctions-test.ts

```diff
@@ -0,0 +1,33 @@
+import { expect } from "chai";
+import { getPercentage } from "../utils/sharedFunctions";
+
+describe("getPercentage", () => {
+  let outOf;
+  beforeEach(() => {
+    outOf = [30, 55, 45, 70];
+  });
+  it("calculates the percentage out of a number", () => {
+    expect(getPercentage(50, 200)).to.equal(25);
+  });
+  it("calculates the percentage out of a string containing a number", () => {
+    expect(getPercentage(50, "200")).to.equal(25);
+  });
+  it("calculates the percentage out of an array of numbers", () => {
+    expect(getPercentage(50, outOf)).to.equal(25);
+  });
+  it("calculates the percentage out of an array of strings containing numbers", () => {
+    outOf = outOf.map(x => `${x}`);
+    expect(getPercentage(50, outOf)).to.equal(25);
+  });
+  it("calculates the percentage out of an array of arrays", () => {
+    // The total is the combined lengths of the sub-arrays; 6, in this case.
+    outOf = [["a"], ["a", "b"], ["a", "b", "c"]];
+    expect(getPercentage(3, outOf)).to.equal(50);
```

Comment:
I don't understand this use-case or where it would come in handy. From the above test, you're using the sum of the numbers in the array, but here you're using the length of arrays in the main array. It doesn't seem consistent.

Reply:
That's exactly how `AggregateList` is currently using it.  It passes in an array of three arrays (one for each production stage) containing library objects, and we need to know how many library objects there are in total.  But I'll try changing that for sake of consistency.